pandas-stubs/io/parsers/readers.pyi

```diff
@@ -0,0 +1,438 @@
+from collections import abc
+import csv
+from types import TracebackType
+from typing import (
+    Any,
+    Callable,
+    Literal,
+    Sequence,
+    overload,
+)
+
+from pandas.core.frame import DataFrame
+from pandas.core.indexes.base import Index
+from pandas.core.series import Series
+
+import pandas._libs.lib as lib
+from pandas._typing import (
+    CompressionOptions,
+    CSVEngine,
+    DtypeArg,
+    FilePath,
+    ReadCsvBuffer,
+    StorageOptions,
+    npt,
+)
+
+from pandas.io.common import IOHandles
+
+@overload
+def read_csv(
+    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
+    *,
+    sep: str | None | lib.NoDefault = ...,
```

Comment:
We shouldn't use `lib.NoDefault` as an option in the stubs.  Users can't use it.  It's there for convenience in the internal typing.

Reply:
Fixed.